server/app/services/applicant/ApplicantService.java

```diff
@@ -1045,4 +1050,118 @@ abstract static class Builder {
       abstract ApplicationPrograms build();
     }
   }
+
+  /** Get corrected address from Esri and formats it as a map compatible with form data */
+  public CompletionStage<ImmutableMap<String, String>> getCorrectedAddress(
+      long applicantId,
+      long programId,
+      String blockId,
+      String selectedAddress,
+      ImmutableList<AddressSuggestion> addressSuggestions) {
+
+    return getReadOnlyApplicantProgramService(applicantId, programId)
+        .thenComposeAsync(
+            roApplicantProgramService -> {
+              Optional<Block> block = roApplicantProgramService.getBlock(blockId);
+
+              if (block.isEmpty()) {
+                return CompletableFuture.failedFuture(
+                    new ProgramBlockNotFoundException(programId, blockId));
+              }
+
+              ApplicantQuestion applicantQuestion =
+                  getFirstAddressCorrectionEnabledApplicantQuestion(block.get());
+              AddressQuestion addressQuestion = applicantQuestion.createAddressQuestion();
+
+              Optional<AddressSuggestion> suggestionMaybe =
+                  addressSuggestions.stream()
+                      .filter(
+                          addressSuggestion ->
+                              addressSuggestion.getSingleLineAddress().equals(selectedAddress))
+                      .findFirst();
+
+              ImmutableMap<String, String> questionPathToValueMap =
+                  buildCorrectedAddressAsFormData(
+                      addressQuestion, suggestionMaybe, selectedAddress);
+
+              return CompletableFuture.completedFuture(questionPathToValueMap);
+            });
+  }
+
+  /** Maps address suggestion and corrected state into a form data compatible map */
+  private ImmutableMap<String, String> buildCorrectedAddressAsFormData(
+      AddressQuestion addressQuestion,
+      Optional<AddressSuggestion> suggestionMaybe,
+      String selectedAddress) {
+
+    Map<String, String> questionPathToValueMap = new HashMap<>(Map.of());
```

Comment:
done

Reply:
Ah sorry, the recommendation is to use the builder instead of `new HashMap<>(Map.of())`:

```suggestion
    ImmutableMap.Builder<String, String> questionPathToValueMap = ImmutableMap.builder();
```

and then below:

```java
    return questionPathToValueMap.build();
```

Reply:
gotcha. done.